sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/MetricExporter.java

```diff
@@ -30,11 +30,11 @@
   /**
    * Exports the collection of {@link MetricData} that have not yet been exported. Note that flush
    * operations can be performed simultaneously depending on the type of metric reader being used.
-   * However, the {@link IntervalMetricReader} will ensure that only one export can occur at a time.
+   * However, the caller MUST ensure that only one export can occur at a time.
    *
    * @return the result of the flush, which is often an asynchronous operation.
    */
-  CompletableResultCode flush();
+  CompletableResultCode forceFlush();
```

Comment:
I'm against renaming this method. It breaks our existing API compatibility guarantees and makes the method out of alignment with the `SpanExporter` interface.  Is there a strong reason why we're adding the "force" to it?

Reply:
I think maybe I'm misunderstanding some comments/conventions here.

I migrated all uses of `flush` to `forceFlush` to conform with SdkTracerProvider.   It looks like exporters use `flush` and the `SDK*Provider` uses `forceFlush`.

I'll revert this specific change.  Do we want SdkMeterProivider to exposes `forceFlush` or just `flush`?  From my perspective each is equally good, so if `flush` was exposed in a stable API, happy to change this.

Reply:
Original comment (I likely took way to far): https://github.com/open-telemetry/opentelemetry-java/pull/3578#discussion_r715309363


For clarity on what we have:

- Exporter has:
  - `flush()`
  - `shutdown()`
- Sdk*Provider has:
  - `forceFlush`
  - `close()`
  - `shutdown()`
  
  If we're on board, I'll make sure metrics matches traces with that.

Reply:
yeah, for good or ill, let's at least have the parallel interfaces have the same methods. I honestly don't remember why exporters got `flush` when the providers got `forceFlush`.